ruby/ql/lib/codeql/ruby/ApiGraphs.qll

```diff
@@ -88,15 +100,49 @@ module API {
      * This predicate may have multiple results when there are multiple constructor calls invoking this API component.
      * Consider using `getAnInstantiation()` if there is a need to distinguish between individual constructor calls.
      */
-    Node getInstance() { result = this.getASubclass().getASuccessor(Label::instance()) }
+    Node getInstance() { result = this.getASubclass().getReturn("new") }
+
+    /**
+     * Gets a node representing the result of calling a method on the receiver represented by this node.
+     */
+    Node getMethod(string method) {
+      result = this.getASubclass().getASuccessor(Label::method(method))
+    }
+
+    /**
+     * Gets a node representing the result of this call.
+     */
+    Node getReturn() { result = this.getASuccessor(Label::return()) }
 
     /**
      * Gets a node representing the result of calling a method on the receiver represented by this node.
      */
-    Node getReturn(string method) {
-      result = this.getASubclass().getASuccessor(Label::return(method))
+    Node getReturn(string method) { result = this.getMethod(method).getReturn() }
+
+    private predicate hasParameterIndex(int n) {
+      exists(string str |
+        exists(this.getASuccessor(Label::parameterByStr(str))) and
+        n = str.toInt()
+      )
+    }
+
+    /** Gets an API node representing the `n`th positional parameter. */
+    Node getParameter(int n) {
+      result = this.getASuccessor(Label::parameter(n)) and this.hasParameterIndex(n)
```

Comment:
👍 

The point was to avoid `InverseAppend` leaking out to the callers of `getParameter` and `getKeywordParameter`, which caused some problems in the JS libraries, so I've also added `pragma[nomagic]` on the two predicates for now.

@erik-krogh's change to IPA type labels was proven to be the most robust fix